### PR TITLE
chore: eslint report-unused-disable-directives

### PR DESCRIPTION
**Pull request by Shinigami92**
https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "docs:serve": "vitepress serve docs --port 5173",
     "docs:diff": "tsx ./scripts/diff.ts",
     "format": "prettier --cache --write .",
-    "lint": "eslint --cache --cache-strategy content .",
+    "lint": "eslint --cache --cache-strategy content --report-unused-disable-directives .",
```
Comment (ST-DDT) [inline package.json:79]: Does this throw or only report?
Is it possible to move this to a config file?

**Review by ST-DDT — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (Shinigami92) [reply package.json:79]: It throws error like any other error thrown by eslint

I did not found a config for that, and I'm the author of eslint-define-config 😅 this arg is totally new to me and I never heard of it until today

Comment (ST-DDT) [reply package.json:79]: There seems to be an option to enable warnings:

- https://github.com/eslint/eslint/issues/15466

Comment (Shinigami92) [reply package.json:79]: So indeed there is a https://eslint.org/docs/latest/use/configure/rules#report-unused-eslint-disable-comments and even my eslint-define-config already has it 😆 

But! it only warns 🙁 
So maybe we can turn on just both?! config for in IDE and cli for PRs

**Review by ST-DDT — APPROVED** [review]
I havent tested this in the IDE but looks good to me.

**Review by xDivisionByZerox — APPROVED** [review]
I like this one.

**Review by ejcheng — APPROVED** [review]

**Pull request merged**